15/umbraco-cms/reference/routing/iisrewriterules.md

```diff
@@ -128,6 +144,49 @@ Another example would be to redirect from non-www to www (except for the Umbraco
 </rule>
 ```
 
-{% hint style="warning" %}
-If you use **Umbraco Cloud** check the [Rewrite Rules](https://docs.umbraco.com/umbraco-cloud/set-up/project-settings/manage-hostnames/rewrites-on-cloud) article.
+### Example: Remove the .aspx Extension
+
+If you want to redirect .aspx URLs to their extensionless counterparts, you can use the following rule. Make sure you also have a web.config file in the root of your site:
+
+```xml
+<configuration>
+  <system.webServer>
+    <rewrite>
+      <rules>
+        <!-- Redirect .aspx URLs to their extensionless counterparts -->
+        <rule name="Remove ASPX extension" stopProcessing="true">
+          <match url="^(.*)\.aspx$" />
+          <conditions logicalGrouping="MatchAll">
+            <add input="{REQUEST_FILENAME}" matchType="IsFile" negate="true" />
+            <add input="{REQUEST_FILENAME}" matchType="IsDirectory" negate="true" />
+          </conditions>
+          <action type="Redirect" url="{R:1}" redirectType="Permanent" />
+        </rule>
+      </rules>
+    </rewrite>
+  </system.webServer>
+</configuration>
+```
+
+### Example: Custom Rewrite Rules for Umbraco Cloud
+
+An example configuration to help ensure your custom rules integrate properly:
+
+```xml
+<?xml version="1.0" encoding="utf-8"?>
+<configuration xmlns:xdt="http://schemas.microsoft.com/XML-Document-Transform">
+  <location path="." inheritInChildApplications="false">
+    <system.webServer>
+      <rewrite xdt:Transform="Insert">
+        <rules>
+          <!-- Add your custom rules here -->
+        </rules>
+      </rewrite>
+    </system.webServer>
+  </location>
+</configuration>
+```
+
+{% hint style="info" %}
+If you use **Umbraco Cloud**, check the [Rewrite Rules](https://docs.umbraco.com/umbraco-cloud/set-up/project-settings/manage-hostnames/rewrites-on-cloud) article.
 {% endhint %}
```

Comment:
I'm not aware whether this is already in the Cloud docs, but in any case, I think it would make more sense over there? 🤔

Reply:
I will add it to the Cloud docs as well. I added it here too cz all the examples mentioned in the article can be used for both CMS and cloud.